app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -7,6 +7,26 @@ def run
           results = SerializableArray.new(results)
           SearchTaskDefinition.create_output(results, output)
         end
+
+        def self.dereference_output(storage, path)
```

Comment:
1. We agree on this one, which is great. I am talking about the most pessimistic case for what you have mentioned - each task supplies its own unique serializer. Of course it doesn't have to do it - it can reuse a generic serializer supplied by the framework that will know how to serialize common objects like AR, hash, array and primitives. We can probably build the serializer extensible enough so it would be possible to add more objects later on.
2. Let's say sender outputs a single instance of SenderOutputProprietaryClass. It's receiver's responsibility to know that it is possible to call `:foo` on it, and it will return a hash with specific keys. In current implementation, Dynflow assumes that the output is a hash, so it tries to "record" all the keys chain that are used by the recipient.
3. I am not sure in which phase dynflow extracts the dependency. If the output is an arbitrary object, it will be very difficult to mimic its behavior. Besides that I am not sure we actually need to... We need only the fact that the recipient is using the sender.
4. Great, as long as task developers can add/modify the serializers.